install_scripts/templates/replicated-operator.sh

```diff
@@ -433,22 +433,23 @@ if [ -z "$PUBLIC_ADDRESS" ] && [ "$AIRGAP" != "1" ] && [ "$NO_PUBLIC_ADDRESS" !=
     printf "Determining service address\n"
     discoverPublicIp
 
+    readReplicatedOperatorOpts "PUBLIC_ADDRESS"
     if [ -z "$PUBLIC_ADDRESS" ]; then
-        readReplicatedOperatorOpts "PUBLIC_ADDRESS"
-        if [ -n "$REPLICATED_OPTS_VALUE" ]; then
-            PUBLIC_ADDRESS="$REPLICATED_OPTS_VALUE"
-            printf "The installer will use service address '%s' (imported from $CONFDIR/replicated-operator 'PUBLIC_ADDRESS')\n" $PUBLIC_ADDRESS
-        fi
+        PUBLIC_ADDRESS="$REPLICATED_OPTS_VALUE"
     fi
-
-    if [ -n "$PUBLIC_ADDRESS" ]; then
-        shouldUsePublicIp
+    if [ -n "$REPLICATED_OPTS_VALUE" ] && [ "$REPLICATED_OPTS_VALUE" = "$PUBLIC_ADDRESS" ]; then
+        PUBLIC_ADDRESS="$REPLICATED_OPTS_VALUE"
```

Comment:
I don't understand how you could be in this block without them already being equal

Reply:
if the public ip from discoverPublicIp line 434 does not match the one from /etc/default/replicated-operator

Reply:
The 2nd test in line 440 checks they're equal so the assignment in 441 is redundant.